src/charm.py

```diff
@@ -283,6 +283,40 @@ def _spring_boot_env(self) -> typing.Dict[str, str]:
             env[JAVA_TOOL_OPTIONS] = jvm_config
         return env
 
+    def _detect_mysql_capability(self) -> bool:
+        """Detect if the mysql lib is present in the spring-boot app.
+
+        Returns:
+            True if the mysql lib is present, False otherwise.
+
+        Raises:
+            TypeError: If the spring-boot app has a not recognizable type.
+        """
+        java_app = self._detect_java_application()
+        container = self._spring_boot_container()
```

Comment:
Can you move this statement to where it is actually used?

Reply:
Code removed in the new version